### DISCUSSION TITLE
Use Ionide.ProjInfo in unit tests

### DISCUSSION THREAD

**Discussion by nojaf** [dotnet/fsharp#13646] [Ideas]
Hello, 

in https://github.com/safesparrow/fsharp/pull/1 I've added a (unit/integration) test to verify the behaviour of the parallelization. 
As @safesparrow correctly points out, constructing the `FSharpProjectOptions` is going to be a churn over time to maintain.

Would the F# team be ok with having [Ionide.ProjInfo](https://github.com/ionide/proj-info) as a test dependency? 
Using [Ionide.ProjInfo.FCS](https://www.nuget.org/packages/Ionide.ProjInfo.FCS) might be less convenient because of the FCS NuGet dependency. 
Copying the contents of https://github.com/ionide/proj-info/blob/main/src/Ionide.ProjInfo.FCS/Library.fs might be easier to go with.

Thoughts? @vzarytovskii @KevinRansom @dsyme @baronfel

**Comment by safesparrow** [comment]
My personal opinion as a complete outsider:

The codebase already requires lots of external libraries and tools, eg. the whole Dotnet SDK.

Could we add a dependency on Ionide.ProjInfo.FCS in a standalone project (either in the codebase or outside) which could then be called out-of-process.

IMO using an improved and polished version of safesparrow/fsharp-benchmark-generator out-of-process would be useful, and could be used in various scenarios.

**Comment by vzarytovskii** [comment]
I'm okay with having it as test dependency, i think we even have it already for some tests.

**Reply by nojaf** [reply]
`FSharp.Compiler.Service.Tests.fsproj` currently has https://www.nuget.org/packages/Dotnet.ProjInfo as dependency.
`Ionide.ProjInfo` is the successor of that.

**Reply by vzarytovskii** [reply]
It will need to be enlisted to our nuget feed then.

**Reply by vzarytovskii** [reply]
Latest version was uploaded to dotnet-public feed

**Reply by nojaf** [reply]
Thanks Vlad!

**Reply by nojaf** [reply]
@vzarytovskii would the [Ionide.ProjInfo.Sln](https://www.nuget.org/packages/Ionide.ProjInfo.Sln/) dependency be on the feed as well?

**Reply by vzarytovskii** [reply]
If it is a dependency of the package, it should be there

**Reply by nojaf** [reply]
Ok, I'm probably a bit impatient by trying it already.
I got:
![image](https://user-images.githubusercontent.com/2621499/183611227-afb70210-f7d1-4fa0-ba75-7c1003dd6e16.png)

Will try again this afternoon.
